pkg/frontend/querymiddleware/roundtrip_test.go

```diff
@@ -516,6 +517,34 @@ func TestRemoteReadMiddleware(t *testing.T) {
 			makeRequest: generateTestRemoteReadRequest,
 			limits:      mockLimits{},
 		},
+		"block an exact query": {
```

Comment:
I'm not fully convinced we should test it here. Why not adding test cases to `Test_queryBlocker_Do`?

Reply:
For end-to-end test, I'll add a test in `integration`